### PR TITLE
Implement signature (SSH) auth

### PR DESCRIPTION
**Pull request by dmach**

### CODE REVIEW

**Comment by dmach** [comment]
This is a showcase of Signature (SSH) auth.
I consider submitting this pull request as a better alternative to spamming #6 with comments.
Feel free to extend this or cherry-pick ideas and throw it away.

File: src/py_obs/osc.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
-import aiohttp
+import base64
```
Comment (dcermak) [inline src/py_obs/osc.py:1]: Imported but unused
```suggestion
```

File: src/py_obs/osc.py [hunk 2]
```diff
@@ -14,16 +22,121 @@ def __str__(self) -> str:
...
+        self.save()
+
+
 @dataclasses.dataclass
 class Osc:
     username: str
     password: str
-
     api_url: str = "https://api.opensuse.org/"
+    ssh_key_path: str = os.path.expanduser("~/.ssh/id_rsa.pub")
```
Comment (dcermak) [inline src/py_obs/osc.py:128]: This shouldn't be hardcoded, I for instance am using my ed25519 key for this.

Comment (dmach) [reply src/py_obs/osc.py]: I've removed the hard-coded value.

File: src/py_obs/osc.py [hunk 3]
```diff
@@ -14,16 +21,126 @@ def __str__(self) -> str:
...
+        self.save()
+
+
 @dataclasses.dataclass
 class Osc:
     username: str
     password: str
-
     api_url: str = "https://api.opensuse.org/"
+    ssh_key_path: str
```
Comment (dcermak) [inline src/py_obs/osc.py:132]: You don't have to specify an SSH key to talk to OBS, only for IBS. Hence this should be optional.
```suggestion
    ssh_key_path: str | None = None
```

File: src/py_obs/osc.py [hunk 4]
```diff
@@ -14,16 +21,126 @@ def __str__(self) -> str:
...
     )
+    _default_headers: tuple[tuple[str, str]] = dataclasses.field(
+        default_factory=lambda: (
+            # https://github.com/openSUSE/open-build-service/issues/13737
+            ("Accept", "application/xml; charset=utf-8"),
+        )
+    )
 
     @staticmethod
     def from_env() -> "Osc":
```
Comment (dcermak) [inline src/py_obs/osc.py:146]: This function will need adjustment as well.

**Pull request merged**